docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+
+### Character types
+
+Faker supports the following character types:
+
+- `CHAR`
+- `VARCHAR`
+- `VARBINARY`
+
+Columns of those types use a generator producing the [Lorem
+ipsum](https://en.wikipedia.org/wiki/Lorem_ipsum) placeholder text. Unbounded
+columns return a random sentence with 3 to 40 words.
+
+To have more control over the format of the generated data, use the `generator`
+column property. Some examples of valid generator expressions:
+
+- `#{regexify '(a|b){2,3}'}`
+- `#{regexify '\\.\\*\\?\\+'}`
+- `#{bothify '????','false'}`
+- `#{Name.first_name} #{Name.first_name} #{Name.last_name}`
+- `#{number.number_between '1','10'}`
+
+See the Datafaker's documentation for more information about
+[the expression](https://www.datafaker.net/documentation/expressions/) syntax
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+To test a generator expression, without having to recreate the table, use the
+`random_string(expression)` function:
+
+```sql
+SELECT random_string('#{Name.first_name}')
+```
+
+### Non-character types
+
+Faker supports the following non-character types:
+
+- `BIGINT`
+- `INTEGER` or `INT`
+- `SMALLINT`
+- `TINYINT`
+- `BOOLEAN`
+- `DATE`
+- `DECIMAL`
+- `REAL`
+- `DOUBLE`
+- `INTERVAL DAY TO SECOND`
+- `INTERVAL YEAR TO MONTH`
+- `TIMESTAMP` and `TIMESTAMP(P)`
+- `TIMESTAMP WITH TIME ZONE` and `TIMESTAMP(P) WITH TIME ZONE`
+- `TIME` and `TIME(P)`
+- `TIME WITH TIME ZONE` and `TIME(P) WITH TIME ZONE`
+- `IPADDRESS`
+- `UUID`
+
+Generator expressions cannot be used for non-character-based columns. To limit
+their data range, specify constraints in the `WHERE` clause.
+
+### Unsupported types
+
+Faker does not support the following data types:
+
+- structural types: `ARRAY`, `MAP`, `ROW`
+- `JSON`
+- Geometry
+- HyperLogLog and all digest types
+
+To generate data using these complex types, data from column of primitive types
+can be combined, like in the following example.
+
+```sql
+CREATE TABLE faker.default.prices (
+  currency VARCHAR NOT NULL WITH (generator = '#{Currency.code}'),
+  price DECIMAL(8,2) NOT NULL
+);
+
+SELECT JSON_OBJECT(KEY currency VALUE price) AS complex
+FROM faker.default.prices
+WHERE price > 0
+LIMIT 3;
+```
+
+Executing these queries should return data structured like this:
+```
+      complex
+-------------------
+ {"TTD":924657.82}
+ {"MRO":968292.49}
+ {"LTL":357773.63}
+(3 rows)
+```
+
+### Number of generated rows
+
+By default, the connector generates 1000 rows for every table. To control how
+many rows are generated for a table, use the `LIMIT` clause in the query. A
+default limit can be set using the `default_limit` table, or schema property or
+in the connector configuration file, using the `faker.default-limit` property.
+
+### Null values
+
+For columns without the `NOT NULL` constraint, null values are generated using
+the default probability of 50%. It can be modified using the `null_probability`
```

Comment:
Hm.. another column property .. maybe we need a list or even table of those at the start of this generator section